src/scancode/cli.py

```diff
@@ -518,21 +528,27 @@ def _scanit(paths, scanners, scans_cache_class, diag,
     return success, rel_path
 
 
-def resource_paths(base_path):
+def resource_paths(base_path, user_ignore):
     """
     Yield tuples of (absolute path, base_path-relative path) for all the files found
     at base_path (either a directory or file) given an absolute base_path. Only yield
     Files, not directories.
     absolute path is a native OS path.
     base_path-relative path is a POSIX path.
-
+    user_ignore- A dictionary, which contains file names, that need to be ignored while scanning
     The relative path is guaranted to be unicode and may be URL-encoded and may not
     be suitable to address an actual file.
     """
+
     base_path = os.path.abspath(os.path.normpath(os.path.expanduser(base_path)))
     base_is_dir = filetype.is_dir(base_path)
     len_base_path = len(base_path)
-    ignored = partial(ignore.is_ignored, ignores=ignore.ignores_VCS, unignores={})
+    user_ignore.update(ignore.ignores_VCS)
+    #Checks if any argument is supplied with --ignore, and if provided, then ignores the files
+    if any(user_ignore):
```

Comment:
Why `if any(user_ignore)` ? `if user_ignore` is plenty enough.
Again the comments are not needed here.